cmd/deploy/deploy.go

```diff
@@ -735,6 +735,17 @@ func (dc *Command) deployDependencies(ctx context.Context, deployOptions *Option
 		if err := dc.PipelineCMD.ExecuteDeployPipeline(ctx, pipOpts); err != nil {
 			return err
 		}
+		if dep.Wait {
+			depBuildEnvs, err := dc.CfgMapHandler.GetDependencyBuildEnvVars(ctx, depName, deployOptions.Namespace)
+			if err != nil {
+				return fmt.Errorf("could not get dependency build env vars: %w", err)
+			}
+			for key, val := range depBuildEnvs {
+				if err := os.Setenv(key, val); err != nil {
```

Comment:
Have you checked if this works for remote deployments?

Reply:
They are passed as part of the Okteto dependency variables. We inject all the variables with the prefix `OKTETO_DEPENDENCY_`. I already tested that is working :)